File: 4/palindrome.jl

# A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

# Find the largest palindrome made from the product of two 3-digit numbers.

function isPalindrome(x)
    if string(x) == reverse(string(x))
        return true
    end
    return false
end

function largest()
    largest = 0
    for i = 100:999
        for j = 100:999
            n = i*j
            if isPalindrome(n) && n > largest
                largest = n
                println(i, ", ", j, ", ", n)
            end
        end
    end
    return largest
end

println("output = ", largest())

File: 4 - Largest palindrome product/palindrome.cpp
#include <iostream>

using namespace std;

int palindrome(int x) {
    string str = to_string(x);
    int l = str.length();
    for (int i = 0; i <= l; i++) {
	if (str[i] != str[l - i - 1]) {
	    return false;
	}
    }
    return true;
}

int main() {
    int n = 0;
    for (int i = 100; i <= 999; i++) {
	for (int j = 100; j <= 999; j++) {
	    if (palindrome(i*j) && i*j > n) {
		n = i*j;
	    }
	}
    }
    printf("output: %d\n", n);
}

